addons/onboarding/models/onboarding_progress_step.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.addons.onboarding.models.onboarding_progress import ONBOARDING_PROGRESS_STATES
+
+
+class OnboardingProgressStep(models.Model):
+    _name = 'onboarding.progress.step'
+    _description = 'Onboarding Progress Step Tracker'
+    _rec_name = 'step_id'
+
+    progress_id = fields.Many2one('onboarding.progress', 'Related Onboarding Progress Tracker', ondelete='cascade')
+    step_state = fields.Selection(
+        ONBOARDING_PROGRESS_STATES, string='Onboarding Step Progress', default='not_done')
+    onboarding_id = fields.Many2one(related='progress_id.onboarding_id', string='Onboarding')
+    step_id = fields.Many2one('onboarding.onboarding.step', string='Onboarding Step', ondelete='cascade')
+
+    _sql_constraints = [
+        ('progress_step_uniq', 'unique (progress_id, step_id)',
+         'There cannot be multiple records of the same onboarding step completion for the same Progress record.'),
+    ]
+
+    def consolidate_just_done(self):
+        was_just_done = self.filtered(lambda progress: progress.step_state == 'just_done')
+        was_just_done.step_state = 'done'
```

Comment:
Would add a "if was_just_done" just before, to avoid useless computation (cache, queries, ...).

Reply:
Seems write does not do anything with a void self now (before there were some additional access checks even on a void recordset).